### PR TITLE
aws-c-event-stream: Use absolute target names

### PR DESCRIPTION
**Pull request by franramirez688**
This PR updates the target names for CMakeDeps via set_property.
Conan version required: 1.43
Conan feature related: conan-io/conan#10099

### CODE REVIEW

File: recipes/aws-c-event-stream/all/conanfile.py [hunk 1]
```diff
@@ -73,12 +73,12 @@ def package(self):
 
     def package_info(self):
         self.cpp_info.set_property("cmake_file_name", "aws-c-event-stream")
-        self.cpp_info.set_property("cmake_target_name", "AWS")
+        self.cpp_info.set_property("cmake_target_name", "AWS::AWS")
```
Comment (SpaceIm) [inline recipes/aws-c-event-stream/all/conanfile.py:76]:
```suggestion
        self.cpp_info.set_property("cmake_target_name", "AWS::aws-c-event-stream")
```

Like `vulkan-headers` recipe, very important to not define `AWS::AWS` target. It was a tedious issue of previous mechanism.

File: recipes/aws-c-event-stream/all/conanfile.py [hunk 2]
```diff
@@ -73,12 +73,12 @@ def package(self):
...
         self.cpp_info.filenames["cmake_find_package"] = "aws-c-event-stream"
         self.cpp_info.filenames["cmake_find_package_multi"] = "aws-c-event-stream"
         self.cpp_info.names["cmake_find_package"] = "AWS"
         self.cpp_info.names["cmake_find_package_multi"] = "AWS"
-        self.cpp_info.components["aws-c-event-stream-lib"].set_property("cmake_target_name", "aws-c-event-stream")
+        self.cpp_info.components["aws-c-event-stream-lib"].set_property("cmake_target_name", "AWS::aws-c-event-stream")
```
Comment (SpaceIm) [inline recipes/aws-c-event-stream/all/conanfile.py:81]:
```suggestion
```

**Review by SpaceIm — CHANGES_REQUESTED** [review]

**Comment by franramirez688** [comment]
@SpaceIm I think we have several PRs with the same conditions as https://github.com/conan-io/conan-center-index/pull/8291#discussion_r760918490
Deleting the component as suggested, we should add all the requires and libs from that one to the root ``cpp_info``, shouldn't we?

**Comment by franramirez688** [comment]
> @SpaceIm I think we have several PRs with the same conditions as [#8291 (comment)](https://github.com/conan-io/conan-center-index/pull/8291#discussion_r760918490) Deleting the component as suggested, we should add all the requires and libs from that one to the root `cpp_info`, shouldn't we?

Checked locally and it's working as it is with legacy and new generators. So it's enough for now.

**Comment by SpaceIm** [comment]
> @SpaceIm I think we have several PRs with the same conditions as [#8291 (comment)](https://github.com/conan-io/conan-center-index/pull/8291#discussion_r760918490)
> Deleting the component as suggested, we should add all the requires and libs from that one to the root `cpp_info`, shouldn't we?

component is not deleted, it's still there as a side effect of cmake_find_package support.

**Review by SpaceIm — APPROVED** [review]

**Review by czoido — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**